### PR TITLE
(WIP): Dropdown

### PR DESCRIPTION
**Pull request by BeniCheni**
Refs #352 

Inspired by [Reach UI - MenuButton (Dropdown)](https://ui.reach.tech/menu-button) & its focus on accessibility, this PR builds on top of the components of Reach UI - MenuButton (Dropdown) to offer: `Dropdown`,`DropdownButton`, `DropdownList`, `DropdownItem`, `DropdownLink`, to support various dropdown compositions.

* Align the versions of deps among packages based on lerna bootstrap hoist warning. Should be warning free now when `npm run prepare` runs lerna bootstrap hoist.

TODO:
- [ ] Write doc
- [ ] Confirm code coverage
- [ ] Need a good way to include global styles, in order to include the base styles from `@reach/menu-button` via .css file.
- [ ] Update stories to improve the concept of Dropdown, versus the current stories still seem `Select`

P.S.: If this PR is successful as a foundation, the next phase of the Dropdown will expand to support composition such as mobile w. fullscreen (similar to the common Popover).

### CODE REVIEW

File: packages/dropdown/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "coverageReporters": [
+      "lcov",
+      "html"
+    ],
+    "testMatch": [
+      "<rootDir>/test/**/*.js"
+    ]
+  },
+  "peerDependencies": {
+    "pcln-design-system": ">=1.x || >=2.x",
```
Comment (jxnblk) [inline packages/dropdown/package.json]: This doesn't look like this is a dependency in the source code...

**Comment by jxnblk** [comment]
According to the Reach docs, it looks like they only offer a `.css` file for including the base styles: https://ui.reach.tech/styling – and we currently don't have a standard way to bundle up CSS files in the design system. In order to use `@reach/menu-button` I think we'll need a good way to include global styles like this in our apps

**Comment by BeniCheni** [comment]
@jxnblk,

> "peerDependencies": {
>    "pcln-design-system": ">=1.x || >=2.x",
> This doesn't look like this is a dependency in the source code...

Good catch! Removed.

>  In order to use @reach/menu-button I think we'll need a good way to include global styles ...

Added a TODO checkbox item in the 1st comment of this PR:
* Need a good way to include global styles, in order to include the base styles from @reach/menu-button via .css file.

Also per another convo about Dropdown stories in this PR being too much like `Select`, I've also added TODO checkbox item:
* Update stories to improve the concept of Dropdown, versus the current stories still seem Select

**Comment by BeniCheni** [comment]
Posted an issue to ask about `style.css` in [#102 in `reach-ui`](https://github.com/reach/reach-ui/issues/102).

**Comment by shakesBeardZ** [comment]
we need this, can some one review it

File: packages/dropdown/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "pcln-dropdown",
+  "version": "2.0.0-beta.7",
+  "description": "React component for dropdowns",
+  "main": "dist/index.js",
+  "scripts": {
+    "prepare": "babel src -d dist",
```
Comment (sdalonzo) [inline packages/dropdown/package.json]: rollup please?

File: packages/dropdown/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "coverageReporters": [
+      "lcov",
+      "html"
+    ],
+    "testMatch": [
+      "<rootDir>/test/**/*.js"
+    ]
+  },
+  "peerDependencies": {
+    "styled-components": ">=3.x || >= 4.x"
```
Comment (sdalonzo) [inline packages/dropdown/package.json]: shouldn't react, react-dom and DS be peers as well?

File: packages/dropdown/src/index.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    value={{
+      identifier: props.identifier
+    }}
+  >
+    <Menu>{props.children}</Menu>
+  </DropdownContext.Provider>
+)
+
+const commonStyle = `
+  font-size: 12px;
```
Comment (sdalonzo) [inline packages/dropdown/src/index.js]: use theme values?

File: packages/dropdown/src/index.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+`
+
+const DropdownLink = styled(MenuLink)`
+  ${color} ${height} ${width} ${commonStyle};
+  display: block;
+  padding: 8px;
+  text-decoration: none;
+`
+
+export { Dropdown, DropdownButton, DropdownList, DropdownItem, DropdownLink }
```
Comment (sdalonzo) [inline packages/dropdown/src/index.js]: should these be attached instead, like Autocomplete and Stepper? ie `Dropdown.Button`, `Dropdown.List`, etc

File: packages/dropdown/test/index.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import renderer from 'react-test-renderer'
+import {
+  Dropdown,
+  DropdownButton,
+  DropdownList,
+  DropdownItem,
+  DropdownLink
+} from '../src'
+
+describe('Dropdown', () => {
```
Comment (sdalonzo) [inline packages/dropdown/test/index.js]: could you add some real tests, instead of just snaps?

**Comment by BeniCheni** [comment]
@shakesBeardZ,

Thanks for your inquiry & interest in the `Dropdown` component. I'll champion a discussion w. the WG this coming session, and come with a path forward for this PR. I the related issue [#102 in reach-ui](https://github.com/reach/reach-ui/issues/102) was also closed w. a suggestion; pending to try it.

I'll update you again in this PR after the convo w. the WG. (cc @wesleymartin85 / @dansternfeld1 / @sdalonzo / @jes708 / @unscsprt )

Meanwhile, feel free to colla. to code review.

**Comment by BeniCheni** [comment]
Unfortunately, after WG session, the feeling is that [`Select`](https://pricelinelabs.github.io/design-system/Select/) is serving most of what a `Dropdown` would offer. There was doubt that if reach-ui or other lib like downshift would be used. Also the underlying native `<select>` on mobile would opt in the mobile browser's UI & benefits for `<select>`.

On that note, this undesired `Dropdown` PR would be closed, as well as its related issue. Further questions could be discussed in the closed issue thought, #352 .

**Pull request closed without merging**